source/MRViewer/MRRibbonMenu.cpp

```diff
@@ -1710,102 +1710,326 @@ void RibbonMenu::setupShortcuts_()
+                    text += "Shift+";
+                }
+
+                if ( key.key == GLFW_KEY_DELETE )
+                {
+                    text += "Delete";
+                }
+                else if ( key.key >= GLFW_KEY_F1 && key.key <= GLFW_KEY_F25 )
+                {
+                    text += "F";
+                    text += std::to_string( key.key - GLFW_KEY_F1 + 1 );
+                }
+                else
+                {
+                    paddings += 2 * cButtonPadding * scaling;
+                }
+
+                float shift = columnWidth - ImGui::CalcTextSize( text.c_str() ).x - paddings;
+                ImGui::SameLine( 0, shift );
+
+                if ( key.mod & GLFW_MOD_CONTROL )
+                {
+                    ImGui::SetCursorPosY( ImGui::GetCursorPosY() - cButtonPadding * scaling );
+                    addReadOnlyLine( std::string( "Ctrl" ) );
+                    ImGui::SameLine( 0, style.ItemInnerSpacing.x );
+                    ImGui::SetCursorPosY( ImGui::GetCursorPosY() - cButtonPadding * scaling );
+                    ImGui::Text( "+" );
+                    ImGui::SameLine( 0, style.ItemInnerSpacing.x );
+                }
+
+                if ( key.mod & GLFW_MOD_ALT )
+                {
+                    ImGui::SetCursorPosY( ImGui::GetCursorPosY() - cButtonPadding * scaling );
+                    addReadOnlyLine( std::string( "Alt" ) );
+                    ImGui::SameLine( 0, style.ItemInnerSpacing.x );
+                    ImGui::SetCursorPosY( ImGui::GetCursorPosY() - cButtonPadding * scaling );
+                    ImGui::Text( "+" );
+                    ImGui::SameLine( 0, style.ItemInnerSpacing.x );
+                }
+
+                if ( key.mod & GLFW_MOD_SHIFT )
+                {
+                    ImGui::SetCursorPosY( ImGui::GetCursorPosY() - cButtonPadding * scaling );
+                    addReadOnlyLine( std::string( "Shift" ) );
+                    ImGui::SameLine( 0, style.ItemInnerSpacing.x );
+                    ImGui::SetCursorPosY( ImGui::GetCursorPosY() - cButtonPadding * scaling );
+                    ImGui::Text( "+" );
+                    ImGui::SameLine( 0, style.ItemInnerSpacing.x );
+                }
+
+                std::string keyStr;
+                if ( key.key >= GLFW_KEY_APOSTROPHE && key.key <= GLFW_KEY_GRAVE_ACCENT )
+                    keyStr += char( key.key );
+                else if ( key.key >= GLFW_KEY_F1 && key.key <= GLFW_KEY_F25 )
+                {
+                    keyStr += "F";
+                    keyStr += std::to_string( key.key - GLFW_KEY_F1 + 1 );
+                }
+                else
+                {
+                    switch ( key.key )
+                    {
+                    case GLFW_KEY_UP:
+                        ImGui::PushFont( fontManager_.getFontByType( RibbonFontManager::FontType::Icons ) );
+                        keyStr += "\xef\x81\xa2";
```

Comment:
better change font scale before push (you can find examples in RibbonButtonDrawer)